### DISCUSSION TITLE
[Question]: What's the alternative for `noBuild`?

### DISCUSSION THREAD

**Discussion by dgonzalezr** [bikecoders/ngx-deploy-npm#591] [Q&A] [answered]
Hello, I've seen that the `noBuild` option has been removed in v8. In our case, we define our publishing process in a Github action as follows:
* 💫 Load and cache dependencies
* ⏳ Build
* 🚀 Publish to NPM registry

We care about building everything before publishing, and we were using the noBuild option to avoid rebuilding when executing the publish command from our Nx monorepo.The publish/deploy Nx target looks like:

```json
    "publish": {
      "executor": "ngx-deploy-npm:deploy",
      "options": {
        "access": "public",
        "noBuild": true,
        "tag": "latest"
      }
    },
```

With the `noBuild` option no longer available, how to avoid rebuilding the project when triggering the publish/deploy command?

**Comment by dianjuar** [comment, marked as answer]
To answer your specific question, I would say that since the internal build process has been removed and you take care of the build process manually before publishing, **remove the `noBuild` option should be enough. Your library will only be built once.**

---
### More info to articulate my answer

The internal build process was deprecated for the Nx feature `dependsOn`. That means that if you want to build before publishing, your target will look like this:

```json
    "publish": {
      "executor": "ngx-deploy-npm:deploy",
      "options": {
        "access": "public",
        "tag": "latest"
      },
     "dependsOn": ["build"]
    },
```

With that configuration, when you execute the `publish` target, it will first build that project. **If you don't want to build your library before executing the `publish` target, remove the `dependsOn`.**

For more information, you can visit `dependsOn` [official documentation](https://nx.dev/reference/project-configuration#dependson)

---

We put much effort into creating migration scripts, thinking in all possible cases to ensure a smooth transition to V8. This specific case is considered, so there is a good chance that the migration scripts have you covered 👍 

```bash
# Run Migrations
nx migrate ngx-deploy-npm
nx migrate --run-migrations

# Review changes 👀
# Commit changes
```

---

This PR covers your case; I hope it will help you through your upgrade. https://github.com/IPS-LMU/octra/pull/187/files

---

Also, this blog post will help users through the ngx-deploy-npm@8.0.0 migration. It talks in depth about all the changes and their motivations.

https://dev.to/dianjuar/whats-coming-to-ngx-deploy-npm-v8-23gn

---

Thank you for bringing your concern.
I wish you success with your upgrade.

Please let us know if you need further help.

**Comment by dgonzalezr** [comment]
Thank you @dianjuar!!! As always, really helpful, I will get back with feedback once I give it a try 🙂

**Comment by dgonzalezr** [comment]
@dianjuar, following your advice, I just upgraded to v8 and triggered a deploy/publish without issue. We can close this now, thank you 🙌🏼
